src/types.ts

```diff
@@ -82,3 +83,11 @@ export enum ChannelTransportType {
 export interface ChannelTransportDesc extends TransportDesc {
   type: ChannelTransportType
 }
+
+export interface ObjectKeeper<T, C, Q> {
+  get(id: string): Promise<T>
+  create?(args: C): Promise<T>
+  find?(attrs?: Q, options?: QueryOptions): Promise<T[]>
+  update?(obj: T): Promise<boolean>
+  delete?(attrs: Q): Promise<boolean>
+}
```

Comment:
where is it going to be used ?

Reply:
Please see: https://github.com/jolocom/sdk-storage-typeorm/pull/19

This would be used for `agent.credentials.get/create/etc`
but also for `agent.interactions` and `agent.identities` eventually!